app/pages/OrganizationListingPage.tsx

```diff
@@ -35,6 +38,18 @@ export const OrganizationListingPage = () => {
   if (!org) { return <Loader />; }
 
   const orgLocations = getResourceLocations(org);
+  const actionBarProps: ActionBarProps = {
+    organization: org,
+    onClickAction: action => {
+      switch (action.icon) {
+        case 'feedback':
+          setFeedbackModalOpen(true);
+          break;
+        default:
+          break;
+      }
```

Comment:
This feels a little bit sketchy, since we have a separate mechanism for specifying the onOnclick handler for actions. What do you think about changing up the structure of the code so that we construct the list of actions in this component and create the feedback action with a `handler` method that calls the `setFeedbackModalOpen()` function, as opposed to constructing the list of actions in `Organization.ts`?

Aside from cleaning up the onClick handler code, I think moving the OrganizationActions here makes more sense than having them in models/Organization.ts, since the actions are really properties of the page, not of the organization itself. For instance, none of the actions directly "modify" the organization as stored in the DB, and half of the actions only make sense in the context of a web page, such as opening the feedback modal, activating the page printing method, or opening up the phone's telephone app via the `tel:` protocol link.

Here's a rough sketch of what I was thinking:

```js
export const OrganizationListingPage = () => {
  const [feedbackModalOpen, setFeedbackModalOpen] = useState(false);
  const allActions = getResourceActions(organization, setFeedbackModalOpen);  // The latter is attached to the feedback action
  const sidebarActions = allActions.filter(a => [/* types here */].include(a.icon));
  const mobileActions = allActions.filter(a => [/* types here */].include(a.icon));

  return (
    /* Actual JSX code where the actions are passed into the appropriate Action component */
  )
}
```

Although it adds more stuff to the page-level component, I think it would simplify the lower-level components.

Reply:
Yeah very good point, this left a sour taste in my mouth but I was reluctant to change too much at once.

The only thing I disagree with in your approach, passing in the setFeedbackModalOpen felt a little unextensible, so I've simplified ActionBar to just take actions[] and onActionClick and emit the action clicked. I removed the `handler` key from action alltogether, so only links are handled by the component itself.

Reply:
Yeah, your new changes here LGTM, I wasn't a huge fan of how my suggestion was passing `setFeedbackModalOpen` into `getResourceActions()`, and your way looks like a better way of accomplishing a similar reduction. Thanks for updating it!